From b7bced77da267e51eaceeef68a213007ba54aa86 Mon Sep 17 00:00:00 2001
From: Matt Turner <mattst88@gmail.com>
Date: Thu, 18 Apr 2024 13:24:18 -0400
Subject: [PATCH] Disable things we don't want

v2: Enable libglad to satisfy egl dependencies
v3: Enable most of libutil to fix undefined references in es2gears
v4: Disable dmabufshare and remove dep on libXext
v5: Drop trackball.c and showbuffer.c
---
 meson.build                   |  3 ++-
 src/egl/opengl/meson.build    | 35 ----------------------------------
 src/egl/opengles2/meson.build |  5 -----
 src/meson.build               |  2 --
 src/util/gl_wrap.h            |  2 --
 src/util/meson.build          |  9 +--------
 src/xdemos/meson.build        | 36 -----------------------------------
 7 files changed, 3 insertions(+), 89 deletions(-)

diff --git a/meson.build b/meson.build
index d487a11f..ff360c2e 100644
--- a/meson.build
+++ b/meson.build
@@ -52,7 +52,7 @@ dep_drm = dependency('libdrm',
   required : get_option('libdrm'),
   disabler : true
 )
-dep_x11 = dependency('x11, xext',
+dep_x11 = dependency('x11',
   required : get_option('x11'),
   disabler : true
 )
@@ -97,6 +97,7 @@ if not dep_glu.found() and host_machine.system() != 'darwin'
   dep_glu = cc.find_library(_glu_name, has_headers: 'GL/glu.h',
     required : dep_x11.found())
 endif
+dep_glu = disabler()
 
 dep_glx = dependency('glx', required: false, disabler : true)
 if not dep_glx.found()
diff --git a/src/egl/opengl/meson.build b/src/egl/opengl/meson.build
index be536280..e7122027 100644
--- a/src/egl/opengl/meson.build
+++ b/src/egl/opengl/meson.build
@@ -20,39 +20,11 @@
 
 _deps = [dep_gl, dep_m, idep_util]
 
-if target_machine.system() == 'linux'
-  executable(
-    'dmabufshare', files('dmabufshare.c'),
-    dependencies: [_deps, idep_glad, idep_eglut],
-    install: true
-  )
-endif
 executable(
   'eglgears', files('eglgears.c'),
   dependencies: [_deps, dep_glu, idep_eglut],
   install: true
 )
-executable(
-  'egltri', files('egltri.c'),
-  dependencies: [_deps, dep_glu, idep_eglut],
-  install: true
-)
-executable(
-  'xeglgears', files('xeglgears.c'),
-  dependencies: [_deps, dep_glu, dep_egl, dep_x11],
-  install: true
-)
-executable(
-  'xeglthreads', files('xeglthreads.c'),
-  dependencies: [_deps, dep_x11],
-  install: true
-)
-
-executable(
-  'eglkms', 'eglkms.c',
-  dependencies: [_deps, dep_drm, dep_gbm, dep_egl],
-  install: true
-)
 
 executable(
   'eglinfo', 'eglinfo.c',
@@ -60,10 +32,3 @@ executable(
   include_directories: [inc_glad],
   install: true
 )
-
-executable(
-  'peglgears', 'peglgears.c',
-  dependencies: [dep_gl, dep_glu, dep_egl, dep_m, idep_util],
-  install: true
-)
-
diff --git a/src/egl/opengles2/meson.build b/src/egl/opengles2/meson.build
index 86062cbd..729c0df6 100644
--- a/src/egl/opengles2/meson.build
+++ b/src/egl/opengles2/meson.build
@@ -29,11 +29,6 @@ executable(
   dependencies: [dep_gles2, idep_eglut, idep_util],
   install: true
 )
-executable(
-  'es2tri', files('es2tri.c'),
-  dependencies: [_deps_x11, dep_m],
-  install: true
-)
 executable(
   'texture_from_pixmap_glesv2', files('texture_from_pixmap_glesv2.c'),
   dependencies: [_deps_x11, idep_util],
diff --git a/src/meson.build b/src/meson.build
index fd4a1673..cea622a6 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -57,5 +57,3 @@ endif
 if host_machine.system() == 'windows'
   subdir('wgl')
 endif
-
-subdir('data')
diff --git a/src/util/gl_wrap.h b/src/util/gl_wrap.h
index b2ff9c8f..f482df5e 100644
--- a/src/util/gl_wrap.h
+++ b/src/util/gl_wrap.h
@@ -7,10 +7,8 @@
 
 #ifdef __APPLE__
 #  include <OpenGL/gl.h>
-#  include <OpenGL/glu.h>
 #else
 #  include <GL/gl.h>
-#  include <GL/glu.h>
 #endif
 
 #ifndef GLAPIENTRY
diff --git a/src/util/meson.build b/src/util/meson.build
index b4fe614f..d2c42dc2 100644
--- a/src/util/meson.build
+++ b/src/util/meson.build
@@ -22,17 +22,10 @@ inc_util = include_directories('.')
 
 files_libutil = files(
   'glinfo_common.c',
-  'readtex.c',
-  'showbuffer.c',
-  'trackball.c',
   'matrix.c',
 )
 
-_deps = [dep_glu, dep_m]
-if dep_glut.found()
-  files_libutil += files('shaderutil.c')
-  _deps += dep_glut
-endif
+_deps = [dep_m]
 
 _libutil = static_library(
   'util',
diff --git a/src/xdemos/meson.build b/src/xdemos/meson.build
index a569fde5..40007616 100644
--- a/src/xdemos/meson.build
+++ b/src/xdemos/meson.build
@@ -21,25 +21,7 @@
 glx_deps = [dep_gl, dep_glx, dep_x11, dep_m]
 
 progs = [
-  'glsync',
-  'glxdemo',
   'glxgears',
-  'glxgears_pixmap',
-  'glxcontexts',
-  'glxheads',
-  'glxpixmap',
-  'glxpbdemo',
-  'glxsnoop',
-  'glxswapcontrol',
-  'manywin',
-  'multictx',
-  'offset',
-  'overlay',
-  'shape',
-  'sharedtex',
-  'texture_from_pixmap',
-  'wincopy',
-  'xfont',
 ]
 foreach p : progs
   executable(
@@ -56,23 +38,7 @@ executable(
   install: true
 )
 
-executable(
-  'xrotfontdemo',
-  files('xrotfontdemo.c', 'xuserotfont.c'),
-  dependencies: glx_deps,
-  install: true
-)
-
-_libpbutil = static_library(
-  'pbutil',
-  files('pbutil.c'),
-  dependencies: glx_deps
-)
-
 pbutil_progs = [
-  'glxgears_fbconfig',
-  'pbinfo',
-  'pbdemo',
 ]
 foreach p : pbutil_progs
   executable(
@@ -84,8 +50,6 @@ foreach p : pbutil_progs
 endforeach
 
 thread_progs = [
-  'glthreads',
-  'sharedtex_mt',
 ]
 foreach p : thread_progs
   executable(
